src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -118,25 +118,21 @@ String getFullPath(String dirPath, String name) {
      * Our objects are small, so I think we can use the maximum batch size without
      * concern for using too much memory.
      */
-    void scanCollectionObjects(Firestore firestore,
+    <V> void scanCollectionObjects(Firestore firestore,
                                String collectionId,
                                int batchSize,
-                               InterruptibleConsumer<QueryDocumentSnapshot> func) throws InterruptedException {
+                               ApiFutureGenerator<V, QueryDocumentSnapshot> generator) throws InterruptedException {
         CollectionReference datasetCollection = firestore.collection(collectionId);
         try {
             int batchCount = 0;
-            int visited;
+            List<QueryDocumentSnapshot> documents;
             do {
-                visited = 0;
                 ApiFuture<QuerySnapshot> future = datasetCollection.limit(batchSize).get();
-                List<QueryDocumentSnapshot> documents = future.get().getDocuments();
+                documents = future.get().getDocuments();
                 batchCount++;
                 logger.info("Visiting batch " + batchCount + " of ~" + batchSize + " documents");
-                for (QueryDocumentSnapshot document : documents) {
-                    func.accept(document);
-                    visited++;
-                }
-            } while (visited >= batchSize);
+                batchOperation(documents, generator);
```

Comment:
The results of `batchOperation()` are discarded, is that OK? I guess `delete()` doesn't return anything interesting if it's successful?

Reply:
Yeah exactly.  I suppose we could use the output to log the number of actual deletes (like if a delete was attempted but there was nothing there which IIRC will just return a `false`)